packages/common/src/store/cache/actions.ts

```diff
@@ -78,24 +66,34 @@ export const addSucceeded = ({
   entries,
   replace = false,
   persist = true
-}: AddSuccededAction) => ({
+}: Omit<AddSuccededAction, 'type'>): AddSuccededAction => ({
   type: ADD_SUCCEEDED,
   kind,
   entries,
   replace,
   persist
 })
 
+type EntriesByKind<EntryT extends Metadata = Metadata> = {
+  [key in Kind]?: Entry<EntryT>[]
+}
```

Comment:
should we move these to a utils file?

if this is likely to be the only usage, prbly okay to leave here. Whichever is more readable